src/Stan/Inspection.hs

```diff
@@ -0,0 +1,47 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+__Inspection__ — check or test provided by Stan.
+-}
+
+module Stan.Inspection
+    ( Inspection (..)
+    , Category (..)
+    , Severity (..)
+
+      -- * Pretty print
+    , prettyShowInspection
+    ) where
+
+import Stan.Core.Id (Id)
+
+
+{- | Data type that represents a check/test, or how we call it
+__inspection__ that is provided by the Stan tool.
+-}
+data Inspection = Inspection
+    { inspectionId          :: !(Id Inspection)
+    , inspectionName        :: !Text
+    , inspectionDescription :: !Text
+    , inspectionSolution    :: !Text
+    , inspectionCategory    :: ![Category]
```

Comment:
Should it be `NonEmpty` list of categories? :thinking: Or is it possible not to have any categories?

Reply:
Oh, nice catch. I wouldn't like to have something uncategorised. This information can be quite valuable and important for filtering and searching. What do you think?